app/models/customer.rb

```diff
@@ -221,5 +231,23 @@ def not_target_opposite_label
 			"not " + not_target_label
 		end
 	end
-  
+
+  private 
+
+  #helper method to calculate and return the length of days in human readable string form 
+  def helper_validity_length(length)
+	if length.nil?
+	 return "(length not set)"
+	end
+   case length
+     when -1
+	then return "(valid for Forever)"
+     when 0..364
+	then return "(valid for #{length} days)"
+     when 365
+	then return "(valid for 1 year)"
+     when 366..730
```

Comment:
One thing that might be more informative is for `when 366..729`, you subtract off 365 to get the number of days over 1 year that is set, so for example, if the number is `367`, it would return `"(valid for 1 year and 2 days)"`, then have another case for `730`, and maybe some other case for > 730 days